core/sql/export/ComDiags.cpp

```diff
@@ -1671,8 +1671,8 @@ ComDiagsArea::ComDiagsArea (CollHeap* ptr): IpcMessageObj(IPC_SQL_DIAG_AREA,0),
 
 ComDiagsArea::ComDiagsArea () :             IpcMessageObj(IPC_SQL_DIAG_AREA,0),
                                             collHeapPtr_(NULL),
-                                            errors_(),
-                                            warnings_(),
+                                            errors_(NULL),
```

Comment:
Yes. 
This is the ComDiagsArea constructor without a heap pointer. There is another one with a heap pointer. I see two options to avoid this leak
a) try to get rid of this constructor OR
b)  Make sure that these NACollection objections are cleared at the end of their lifecycle.

Either of them would take some effort. I don't have a fix that can be included now. I would claim that this code does no worse than before here, while making marginal improvements in other places. It also ensures that no one unintentionally creates objects on the system heap.